### ISSUE TITLE
vue3 example/tooling broken for me

### ISSUE THREAD

**Issue by adq** [single-spa/single-spa-vue#71]
Hi, I'm very new to single-spa-vue, but I think I've maybe spotted an issue with the vue3 example **and the create-single-spa tooling**.

On https://single-spa.js.org/docs/ecosystem-vue/, it says this:

`
[....]
const vueLifecycles = singleSpaVue({
  createApp,
  appOptions: {
    render() {
      return h(App, {
        // single-spa props are available on the "this" object. Forward them to your component as needed.
        // https://single-spa.js.org/docs/building-applications#lifecyle-props
        name: this.name,
        mountParcel: this.mountParcel,
        singleSpa: this.singleSpa,
      });
    },
  },
  handleInstance: (app) => {
    app.use(router);
  }
});
[...]
`

However, if I do that, I get the following failure when it tries to mount the component:
`
app-errors.js:12 Uncaught TypeError: application '@tvsquared/tvs-spapoc-nav' died in status SKIP_BECAUSE_BROKEN: Cannot convert object to primitive value
    at patchAttr (runtime-dom.esm-bundler.js?830f:168)
    at patchProp (runtime-dom.esm-bundler.js?830f:354)
    at mountElement (runtime-core.esm-bundler.js?5c40:3826)
    at processElement (runtime-core.esm-bundler.js?5c40:3789)
    at patch (runtime-core.esm-bundler.js?5c40:3709)
    at componentEffect (runtime-core.esm-bundler.js?5c40:4211)
    at reactiveEffect (reactivity.esm-bundler.js?a1e9:42)
    at effect (reactivity.esm-bundler.js?a1e9:17)
    at setupRenderEffect (runtime-core.esm-bundler.js?5c40:4176)
    at mountComponent (runtime-core.esm-bundler.js?5c40:4134)
`

Digging in shows that it patchAttr is trying to set the values name, mountParcel, and singleSpa on the DOM element, which it can't do 'cos singleSpa is an object so cannot be converted to a primitive value.

I'm not sure what the correct way to fix this is; if I omit 
`
        name: this.name,
        mountParcel: this.mountParcel,
        singleSpa: this.singleSpa,
`

completely, things work, but I don't know what effect that will have.

**Comment by adq** [comment]
Update! Aahhhhh, so if I define

```
  props: {
    name: String,
    mountParcel: Function,
    singleSpa: Object
  }
```

In my App.vue, it stops trying to render them into the DOM. Right ok, that makes sense.

Why are those in the example for vue3 and not vue2? I'd suggest removing them from the vue3 one as well maybe.

**Comment by joeldenning** [comment]
Thanks for the description here - I think the best solution would be to modify App.vue to add those props. However, that can be tricky, as App.vue is an existing file when vue-cli-plugin-single-spa runs and it's hard to parse and modify existing files.

I don't think that deleting the props is a good solution, as it's a very common question we receive "how do I use single-spa props." Perhaps we could comment out the props in the main file, explaining that you can uncomment the props if you modify the App.vue file accordingly. The reason that the props exist in the Vue 3 main file but not the Vue 2 file is that we added the props because it was so common of a question, and forgot to do it in both places.

I'm transferring this to vue-cli-plugin-single-spa. Here's the relevant file: https://github.com/single-spa/vue-cli-plugin-single-spa/blob/89d9868f1b3ff9db1870452e23586cd78f89e503/template/src/main-vue-3.js#L14. Do you have interest in submitting a pull request commenting out the props in vue 3, and adding the commented props to vue 2?
